### PR TITLE
Implement request index page bottom part ui

### LINKED ISSUE #859: Implement request index page bottom part UI

### PR DESCRIPTION
**Pull request by prin-r**
Fix #859

### CODE REVIEW

File: scan/src/pages/RequestIndexPage.re [hunk 1]
```diff
@@ -119,10 +134,280 @@ let make = (~reqID, ~hashtag: Route.request_tab_t) => {
...
+                 weight=Text.Semibold
+                 spacing={Text.Em(0.06)}
+                 color=Colors.gray6
+               />
+             </Col>
+             <Col size=5.>
+               <ProgressBar
+                 reportedValidators=numReport
+                 minimumValidators={request.sufficientValidatorCount}
+                 totalValidators
```
Comment (evilpeach) [inline scan/src/pages/RequestIndexPage.re]: Font Size should be 10px of Pending 3 Validators

File: scan/src/rpc/ProofHook.re [hunk 2]
```diff
@@ -12,6 +12,8 @@ module Proof = {
 };
 
 let get = (requestId: int) => {
-  let json = AxiosHooks.use({j|bandchain/proof/$requestId|j});
+  // Mock
+  let json =
```
Comment (evilpeach) [inline scan/src/rpc/ProofHook.re]: Let's use the real one.

File: scan/src/components/KVTable.re [hunk 3]
```diff
@@ -1,91 +1,225 @@
+type field_t =
+  | Value(string)
+  | Values(list(string))
+  | DataSource(ID.DataSource.t, string)
+  | Block(ID.Block.t)
+  | TxHash(Hash.t);
+
+type theme_t =
+  | THEME_1
+  | THEME_2;
```
Comment (evilpeach) [inline scan/src/components/KVTable.re]: I think we should name it meaningful.

File: scan/src/components/TypeID.re [hunk 4]
```diff
@@ -28,7 +31,8 @@ module Styles = {
     | Title => style([pointerEvents(`none)])
     | Landing
     | Subtitle
-    | Text => style([pointerEvents(`auto)]);
+    | Text => style([pointerEvents(`auto)])
+    | Mini => style([pointerEvents(`auto)]);
```
Comment (evilpeach) [inline scan/src/components/TypeID.re]: It should be 
```
| Text
| Mini => style([pointerEvents(`auto)]);
```

**Review by evilpeach — CHANGES_REQUESTED** [review]
As I go through, I see the Progress Bar works buggy.
For example
image

The requested validators are 3 but the indicator doesn't point at the end of progress bar.

File: scan/src/components/KVTable.re [hunk 5]
```diff
@@ -1,91 +1,232 @@
...
+      <TypeID.Block id position=TypeID.Mini />
+    </div>
+  | TxHash(txHash) =>
+    <div className={Styles.valueContainer(maxWidth)}>
+      <TxLink txHash width=maxWidth size=Text.Sm />
+    </div>
+  };
 };
 
+let withSetting = (arr, sizes, isRights) =>
```
Comment (sorawit) [inline scan/src/components/KVTable.re:156]: Make it more flexible.

**Review by sorawit — DISMISSED** [review]

File: scan/src/pages/RequestIndexPage.re [hunk 6]
```diff
@@ -119,10 +132,280 @@ let make = (~reqID, ~hashtag: Route.request_tab_t) => {
...
+                  }}
+               </div>
+             </Col>
+           </div>
+           <VSpacing size=Spacing.sm />
+           <div className={Styles.topicContainer(40)}>
+             <Col size=1.>
+               <div className=Styles.hFlex>
+                 <Text
+                   value="EXPIRATION DATA"
```
Comment (taobun) [inline scan/src/pages/RequestIndexPage.re]: External Data

File: scan/src/pages/RequestIndexPage.re [hunk 7]
```diff
@@ -119,10 +132,280 @@ let make = (~reqID, ~hashtag: Route.request_tab_t) => {
...
+                   size=Text.Sm
+                   weight=Text.Semibold
+                   spacing={Text.Em(0.06)}
+                   color=Colors.gray6
+                 />
+                 <HSpacing size=Spacing.md />
+                 // Mock
+                 <CopyButton data={"aaaa" |> JsBuffer.fromHex} />
```
Comment (taobun) [inline scan/src/pages/RequestIndexPage.re]: Remove this button.

File: scan/src/components/KVTable.re [hunk 8]
```diff
@@ -1,91 +1,232 @@
...
+    {
+      mainElem: elem,
+      size: sizes->Belt_List.get(i)->Belt_Option.getWithDefault(1.),
+      isRight: isRights->Belt_List.get(i)->Belt_Option.getWithDefault(false),
+    }
+  );
+
 [@react.component]
-let make = (~header=["KEY", "VALUE"], ~kv) => {
+let make =
```
Comment (taobun) [inline scan/src/components/KVTable.re:166]: Can you switch theme for header table font size

**Review by taobun — CHANGES_REQUESTED** [review]

File: scan/src/pages/RequestIndexPage.re [hunk 9]
```diff
@@ -4,8 +4,16 @@ module Styles = {
   let pageContainer = style([paddingTop(`px(20))]);
 
   let vFlex = style([display(`flex), flexDirection(`row), alignItems(`center)]);
+  let hFlex = style([display(`flex), alignItems(`center)]);
 
-  let logo = style([width(`px(50)), marginRight(`px(10))]);
+  let centerHFlex = style([display(`flex), alignItems(`center), justifyContent(`center)]);
+
+  let minWidth = mw => style([minWidth(`px(mw)), maxWidth(`px(mw))]);
```
Comment (evilpeach) [inline scan/src/pages/RequestIndexPage.re]: Why don't we set width instead?

File: scan/src/pages/RequestIndexPage.re [hunk 10]
```diff
@@ -4,8 +4,16 @@ module Styles = {
   let pageContainer = style([paddingTop(`px(20))]);
 
   let vFlex = style([display(`flex), flexDirection(`row), alignItems(`center)]);
+  let hFlex = style([display(`flex), alignItems(`center)]);
 
-  let logo = style([width(`px(50)), marginRight(`px(10))]);
+  let centerHFlex = style([display(`flex), alignItems(`center), justifyContent(`center)]);
```
Comment (evilpeach) [inline scan/src/pages/RequestIndexPage.re]: Let's make `let center = style([justifyContent(center)])` and then use `Css.merge` to combine `hFlex` and `center` together.

**Review by evilpeach — CHANGES_REQUESTED** [review]

File: scan/src/components/KVTable.re [hunk 11]
```diff
@@ -1,91 +1,237 @@
...
+    </div>
+  | DataSource(id, name) =>
+    <div className={Styles.valueContainer(maxWidth)}>
+      <TypeID.DataSource id position=TypeID.Mini />
+      <HSpacing size=Spacing.sm />
+      <Text
+        value=name
+        weight=Text.Regular
+        spacing={Text.Em(0.02)}
+        code=true
```
Comment (evilpeach) [inline scan/src/components/KVTable.re]: remove the code props

**Review by evilpeach — APPROVED** [review]
LGTM

File: scan/src/rpc/RequestHook.re [hunk 12]
```diff
@@ -114,10 +114,78 @@ module Request = {
...
+    values: [
+      {externalDataID: 2, data: "6c756c75" |> JsBuffer.fromHex},
+      {externalDataID: 4, data: "6c756c75" |> JsBuffer.fromHex},
+      {externalDataID: 1, data: "6c756c75" |> JsBuffer.fromHex},
+      {externalDataID: 3, data: "6c756c75" |> JsBuffer.fromHex},
+    ],
+  },
+];
+
+let get: 'a => option(Request.t) =
```
Comment (taobun) [inline scan/src/rpc/RequestHook.re:157]: This file will be removed, so it's ok to land with this

**Review by taobun — APPROVED** [review]

**Pull request merged**